### ISSUE TITLE
Toolchain override

### ISSUE THREAD

**Issue by theduke** [sagiegurari/cargo-make#6]
Similar to the platform override, it would be very nice to have a toolchain override for stable / beta / nightly.

This is most useful for running additional stuff on nightly, or ignoring things on nightly.

This can of course be already done with a condition_script, but that's pretty awkward compared to the override, Especially since you need a different script for windows and linux.

Even better would be a combination with the platform.

So, something like this:

```toml
[tasks.build]
...

[tasks.build.nightly]
...

[tasks.build.windows.nightly]
...
```

**Comment by sagiegurari** [comment]
I want to say that it over complicates things but I think that rust channel is important enough to implement and might be a common use case.
So I'll try to get it done in the coming days.

**Comment by theduke** [comment]
An alternative solution might be to allow specifying conditions on a task.

That would not enable you to override a task, but for me, skipping would often be enough.

Example:

```toml
[tasks.whatever]
condition = { channel = ["stable", "beta"], platform = "windows" }
```

**Comment by sagiegurari** [comment]
That looks better. I'll probably play with that a bit more to be more extendable.

**Comment by sagiegurari** [comment]
Added the new condition attribute as you recommended.
You can see the [condition docs](https://github.com/sagiegurari/cargo-make#usage-conditions-structure)
It was a good idea that will enable me in the future to add more types of conditions like checking env var exists or if env var is equal to some value and so on...

Please check if it meets you needs

**Comment by sagiegurari** [comment]
Added 3 more condition types: env (checks env is set and equals to some value), env_set (just ensures env var is defined), env_not_set (reverse of env_set which ensures an env var is not defined).
All 5 condition types are documented [here](https://github.com/sagiegurari/cargo-make#usage-conditions-structure)

Hope this provides all the needed capabilites.
For me, I was able to remove all condition_scripts from the default toml and replace them with this new condition attribute which basically also helps make those task definitions more multi platform.
